src/Domain/CoordinateMaps/TimeDependent/Rotation3D.cpp

```diff
@@ -0,0 +1,220 @@
+      GetRotationMatrixDeriv(time, functions_of_time, f_of_t_name_);
+
+  std::array<tt::remove_cvref_wrap_t<T>, 3> result;
+  for (size_t i = 0; i < 3; i++) {
+    result[i] = rot_matrix_deriv(i, 0) * source_coords[0];
+    for (size_t j = 1; j < 3; j++) {
+      result[i] += rot_matrix_deriv(i, j) * source_coords[j];
+    }
+  }
+  return result;
+}
+
+template <typename T>
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> Rotation3D::jacobian(
+    const std::array<T, 3>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const noexcept {
+  ASSERT(functions_of_time.find(f_of_t_name_) != functions_of_time.end(),
+         "Could not find function of time: '"
+             << f_of_t_name_ << "' in functions of time. Known functions are "
+             << keys_of(functions_of_time));
+
+  const Matrix rot_matrix =
+      GetRotationMatrix(time, functions_of_time, f_of_t_name_);
+
+  // Make tensor of zeros with correct type
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> jacobian_matrix{
+      make_with_value<tt::remove_cvref_wrap_t<T>>(
+          dereference_wrapper(source_coords[0]), 0.0)};
+  // The jacobian is just the rotation matrix
+  for (size_t i = 0; i < 3; i++) {
+    for (size_t j = 0; j < 3; j++) {
+      jacobian_matrix.get(i, j) = rot_matrix(i, j);
+    }
+  }
+
+  return jacobian_matrix;
+}
+
+template <typename T>
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame>
+Rotation3D::inv_jacobian(
+    const std::array<T, 3>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const noexcept {
+  ASSERT(functions_of_time.find(f_of_t_name_) != functions_of_time.end(),
+         "Could not find function of time: '"
+             << f_of_t_name_ << "' in functions of time. Known functions are "
+             << keys_of(functions_of_time));
+
+  const Matrix rot_matrix =
+      GetRotationMatrix(time, functions_of_time, f_of_t_name_);
+
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> inv_jacobian_matrix{
+      make_with_value<tt::remove_cvref_wrap_t<T>>(
+          dereference_wrapper(source_coords[0]), 0.0)};
+  // The inverse jacobian is just the inverse rotation matrix, which is the
+  // transpose of the rotation matrix.
+  for (size_t i = 0; i < 3; i++) {
+    for (size_t j = 0; j < 3; j++) {
+      inv_jacobian_matrix.get(i, j) = rot_matrix(j, i);
+    }
+  }
+
+  return inv_jacobian_matrix;
+}
+
+void Rotation3D::pup(PUP::er& p) noexcept {
+  p | f_of_t_name_;
+}
+
+bool operator==(const Rotation3D& lhs, const Rotation3D& rhs) noexcept {
+  return lhs.f_of_t_name_ == rhs.f_of_t_name_;
+}
+
+bool operator!=(const Rotation3D& lhs, const Rotation3D& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+#define DIM(data) BOOST_PP_TUPLE_ELEM(0, data)
```

Comment:
Replace `DIM(data)` with 3 everywhere.  There's no way this class will ever work for a different value.